rviz_common/CMakeLists.txt

```diff
@@ -280,6 +283,7 @@ ament_export_dependencies(
   tf2_geometry_msgs
   tf2_ros
 )
+ament_export_include_directories(include)
```

Comment:
What I don't understand is why this suddenly becomes necessary. If I only build dynamically, the call to `ament_export_interfaces(rviz_common)` is enough (although the `rviz_common_INCLUDE_DIRS` are not properly populated), but once I try to build the object files, I can't find include files from `rviz_rendering`. 
I can fix this here no problem, but I have the same issue with resource_retriever, where I can't add the call to `ament_export_include_directories`.
Sorry to bother you @wjwwood , but can you explain this phenomenon?

Reply:
`ament_export_interfaces(rviz_common)` exports the cmake target `rviz_common`, so when using it in another package (as `rviz_common::rviz_common`), it brings with it the include dirs, link flags, etc...

In this mode of operation (target exporting) all settings are target specific, so things like `rviz_common_INCLUDE_DIRS` are not used.

If you want that directory to be included in the `rviz_common` target you need some `target_include_directories` magic, see:

https://github.com/bsinno/rviz/blob/6f27b4dfa42b1adc6c8692fd866600fb5282b32d/rviz_common/CMakeLists.txt#L236-L242

Which is already happening, so you should only need to link what ever you're compiling that needs those headers against `rviz_common::rviz_common` to get the right stuff.

Reply:
Thank you for the explanation. 
That's actually a problem, because if you compile OBJECT libraries like I do in `rviz_default_plugins` (to later compile to shared libraries for the real application and to a static library to link against tests), you can't specify link targets. 

More specifically: In CMakeLists of `rviz_default_plugins`, I'm doing
```
add_library(rviz_default_plugins_object OBJECT
  ${rviz_default_plugins_headers_to_moc}
  ${rviz_default_plugins_source_files}
)
```
If I don't have the line above, I can't find include directories of `resource_retriever` specified via `target_include_directories(rviz_default_plugins_object PUBLIC ${rviz_common_INCLUDE_DIRS})`
what I'd have to do is say
`target_link_libraries(rviz_default_plugins_object PUBLIC rviz_common::rviz_common)`
or 
`ament_target_dependencies(rviz_default_plugins_object rviz_common)`.
However, neither is possible, because you mustn't link against OBJECT libraries.
That explains why I need a `ament_export_include_directories`. For `rviz_common` that's no problem, for `resource_retriever` I can't do this, but I asked whether this could be added (https://github.com/ros/resource_retriever/issues/21).